models/nc_news.models.js

```diff
@@ -109,3 +110,28 @@ exports.fetchAllUsers = () => {
     return rows;
   });
 };
+
+exports.checkIfArticleExists = (article_id) => {
+  return db
+    .query(
+      `SELECT * FROM articles 
+      WHERE article_id = $1`,
+      [article_id]
+    )
+    .then(({ rows: articles }) => {
+      if (articles.length === 0) {
+        return Promise.reject({ status: 404, message: "Not found." });
+      }
+    });
+};
+
+exports.checkIfTopicExists = (topic) => {
+  return db
+    .query(`SELECT * FROM topics WHERE slug = $1`, [topic])
+    .then(({ rows: topic }) => {
+      if (topic.length === 0) {
+        return Promise.reject({ status: 404, message: "Not found" });
+      }
+      return [];
+    });
+};
```

Comment:
Well done for checking the topic exists, and not hardcoding a green list of topics. There is a later task to post topics, so this is in preparation for that :)